src/serialization/compact/SchemaService.ts

```diff
@@ -74,11 +89,19 @@ export class SchemaService {
             this.logger.trace('SchemaService', `Schema id ${schemaId} already exists locally`);
             return Promise.resolve();
         }
-        const message = ClientSendSchemaCodec.encodeRequest(schema);
-        const invocation = new Invocation(this.getInvocationService(), message);
-        return this.getInvocationService().invoke(invocation).then(() => {
-            this.putIfAbsent(schema);
-        });
+        return this.replicateSchemaInCluster(schema).then((result) => {
+            if (!result) {
+                throw new IllegalStateError(
+                    `The schema ${schema.typeName} cannot be replicated in the cluster, after ${this.maxPutRetryCount}  retries. 
+                    It might be the case that the client is connected to the two halves of the cluster that 
+                    is experiencing a split-brain, and continue putting the data associated with that schema might 
+                    result in data loss. It might be possible to replicate the schema after some time, when 
+                    the cluster is healed.`
+                );
+            } else {
+                this.putIfAbsent(schema);
+            }
+        }).catch((err) => { console.error(err); });
```

Comment:
We don't use console.error directly. We can either ignore the error completely, by `catch(() => {})` or if we are sure that the error handling of the promise returned by this method will be handled, we can skip putting a catch(). We can also log the error if we want via the logger.  

Right now, all the register errors are propagated to the user via async APIs. If the user doesn't register an error handler the program may crash if the user is using a new Node.js runtime. I think we can remove the catch block for this promise since we cannot continue without a schema. User should get an error.

Reply:
I agree with you. I forgot to remove catch block. I only use for testing. Thanks for attention.

I removed catch block.